icu4c/source/common/localematcher.cpp

```diff
@@ -605,10 +605,10 @@ class LocaleLsrIterator {
 
 const Locale *LocaleMatcher::getBestMatch(const Locale &desiredLocale, UErrorCode &errorCode) const {
     if (U_FAILURE(errorCode)) { return nullptr; }
-    int32_t suppIndex = getBestSuppIndex(
+    auto [suppIndex, found] = getBestSuppIndex(
```

Comment:
Here it might be me who's stuck in the old ways, returning `std::pair<value, status>` is what the standard library does because it didn't use to have `std::optional` but we have that now and that also maps 1-to-1 to Java so I've now updated the code to use that instead.